### PR TITLE
Curry documentation and tests

### PR DESCRIPTION
**Pull request by mariari**
Adds documentation to curry, and creates minor tests for it

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -257,31 +263,31 @@ In particular,
...
-1. The construction of binary products $A × B$ of sets $A,B$, and the empty product $\mathsf{1}$.
+1. The construction of binary products $A × B$ of sets $A,B$, and the empty product $mathsf{1}$.
 
 2. The construction of “function spaces” $B^A$ of sets $A,B$, called *exponentials*,
    i.e., collections of functions between pairs of sets.
 
 3. The so-called [*currying*](https://en.wikipedia.org/wiki/Currying)
 of functions,
-   $C^{(B^A)} \cong C^{(A × B)}$,
+   $C^{(B^A)} cong C^{(A × B)}$,
```
Comment (agureev) [inline README:273]: Maybe it is better to instead use $Hom(A, C^B) cong Hom(A x B, C)$ following our recent discussion

File: README [hunk 2]
```diff
@@ -1241,6 +1249,30 @@ Various utility functions ontop of @GEB-CATEGORIES
...
+    
+    where c^b means c to the exponent of b (EXPT c b)
+    
+    ```
+    Γ, f : a × b → c,
+    --------------------
+    (curry f) : a → c^b
+    ```
+    
+    In category terms, `a → c^b` is isomorphic to `a → b → c`
```
Comment (agureev) [inline README:1273]: Again, might be a good idea to change up the notation a bit

**Review by agureev — APPROVED** [review]

File: README (hunk 1 above)
Comment (mariari) [reply README:273]: This doesn't seem changed in this commit, probably should have a follow up pr for this

File: README (hunk 2 above)
Comment (mariari) [reply README:1273]: perhaps, though I think here this gives a more intuitive feeling for it?

We should have a followup pr on this issue as well

**Pull request merged**